File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    let i = Number.MAX_VALUE;
    let j = Number.MAX_VALUE;
    for(const k of nums)
        if (k <= i) i = k;
        else if (k <= j) j = k;
        else return true;
    return false;
    
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
      bool increasingTriplet(vector<int>& nums) {
        int min1 = INT_MAX;  
        int min2 = INT_MAX;  
        for (int min3 : nums)
            if (min3 <= min1) 
                min1 = min3; 
            else if (min3 <= min2)
                min2 = min3;  
            else 
                return true;
        return false; 
    }
};